addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -311,6 +311,10 @@ var SnippetEditor = Widget.extend({
         // If it is an invisible element, we must close it before deleting it (e.g. modal)
         await this.toggleTargetVisibility(!this.$target.hasClass('o_snippet_invisible'));
 
+        this.trigger_up('widgets_stop_request', {
+            $target: this.$target,
+        });
```

Comment:
Even if the editor team broke that rule at some places I think, the "widget start and stop" notions are not supposed to be something the editor knows about.

As an example, see how widgets are destroyed before saving: `ready_to_clean_for_save`